ee/tabby-webserver/src/db.rs

```diff
@@ -0,0 +1,132 @@
+use std::{env, path::PathBuf, sync::Arc};
+
+use anyhow::Result;
+use lazy_static::lazy_static;
+use rusqlite::params;
+use rusqlite_migration::{AsyncMigrations, M};
+use tokio_rusqlite::Connection;
+
+lazy_static! {
+    static ref MIGRATIONS: AsyncMigrations = AsyncMigrations::new(vec![M::up(
+        r#"
+            CREATE TABLE IF NOT EXISTS register_token (
+                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                token VARCHAR(255) NOT NULL,
+                created_at TIMESTAMP DEFAULT (DATETIME('now')),
+                updated_at TIMESTAMP DEFAULT (DATETIME('now')),
+                CONSTRAINT `idx_token` UNIQUE (`token`)
+            );
+        "#
+    ),]);
+    static ref TABBY_ROOT: PathBuf = match env::var("TABBY_ROOT") {
+        Ok(x) => PathBuf::from(x),
+        Err(_) => PathBuf::from(env::var("HOME").unwrap()).join(".tabby"),
+    };
+}
+
+fn db_file() -> PathBuf {
+    TABBY_ROOT.join("db.sqlite3")
+}
+
+pub struct DbConn {
+    conn: Arc<Connection>,
+}
+
+impl DbConn {
+    pub async fn new() -> Result<Self> {
+        let conn = Connection::open(db_file()).await?;
+        Self::init_db(conn).await
+    }
+
+    /// Initialize database, create tables and insert first token if not exist
+    async fn init_db(mut conn: Connection) -> Result<Self> {
+        MIGRATIONS.to_latest(&mut conn).await?;
+
+        let token = uuid::Uuid::new_v4().to_string();
+        conn.call(move |c| {
+            c.execute(
+                r#"INSERT OR IGNORE INTO register_token (id, token) VALUES (1, ?)"#,
```

Comment:
```suggestion
                r#"INSERT OR IGNORE INTO registeration_token (id, token) VALUES (1, ?)"#,
```

ah, sorry, i didn't notice i used a inconsistent naming between here and `reset_registeration_token`, fixed.

Reply:
No worries, anyway, I'll use term `registration_token` for table name, methods, SQL, etc.